src/fastcs_eiger/eiger_controller.py

```diff
@@ -83,7 +85,12 @@ async def put(
             print(
                 f"Fetching parameters after setting {self.uri}: {parameters_to_update}"
             )
-
+        for param in FETCH_BEFORE_RETURNING:
+            if param in parameters_to_update:
+                print(f"Fetching parameter {param} before returning from put")
+                parameters_to_update.remove(param)
+                attr = controller.attributes.get(_key_to_attribute_name(param))
+                await attr.updater.config_update(controller, attr)
         await controller.queue_update(parameters_to_update)
```

Comment:
Nit: I would visually separate these steps

```suggestion
                await attr.updater.config_update(controller, attr)
 
        await controller.queue_update(parameters_to_update)
```